test/Bitmap/IndexedBmpWriter.test.cpp

```diff
@@ -12,8 +12,8 @@ TEST(BitmapFile, InvalidBitCountThrows)
 	const std::vector<uint8_t> pixels(4, 0);
 	const std::string filename = "Sprite/data/MonochromeTest.bmp";
 
-	EXPECT_THROW(BitmapFile::WriteIndexed(filename, 3, 1, 1, palette, pixels), std::runtime_error);
-	EXPECT_THROW(BitmapFile::WriteIndexed(filename, 32, 1, 1, palette, pixels), std::runtime_error);
+	EXPECT_THROW(BitmapFile::CreateIndexed(3, 1, 1, palette, pixels).WriteIndexed(filename), std::runtime_error);
```

Comment:
Most of these tests are writing to disc during their tests. I suspect I wrote these tests before the `Stream::DynamicMemoryReader` class existed. I'd like to edit the majority of these tests to write to memory dynamically in a future branch. Maybe leave one to test file writing. The test code should become cleaner and maybe run faster.

Reply:
I approve of in memory tests. Faster, and less likely to leave crud around on disk, especially if the unit tests fail in the middle of a test. That has the potential to impact future test runs, in case they are sensitive to files already existing, or having the wrong write permissions or user ownership.

Might make sense to assume stream writing to disk is tested by the stream unit tests, and just assume it is good here. No need to test stream code in the bitmap tests.